File: 13-work-with-basic-auth/java/src/test/java/WorkWithBasicAuth2.java
package src.test.java;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.By;
import org.openqa.selenium.firefox.FirefoxDriver;
import static org.hamcrest.CoreMatchers.*;
import static org.hamcrest.MatcherAssert.assertThat;

public class WorkWithBasicAuth2 {
    WebDriver driver;

    @Before
    public void setUp() throws Exception {
        driver = new FirefoxDriver();
        driver.get("http://admin:admin@the-internet.herokuapp.com/basic_auth");
    }

    @After
    public void tearDown() throws Exception {
        driver.quit();
    }

    @Test
    public void workWithBasicAuthTest2() {
        driver.get("http://the-internet.herokuapp.com/basic_auth");
        String pageMessage = driver.findElement(By.cssSelector("p")).getText();
        assertThat(pageMessage, containsString("Congratulations!"));
    }

}

File: 13-work-with-basic-auth/kotlin/src/test/kotlin/WorkWithBasicAuth2.kt
import org.openqa.selenium.By
import org.openqa.selenium.firefox.FirefoxDriver
import kotlin.test.Test
import kotlin.test.BeforeTest
import kotlin.test.AfterTest
import kotlin.test.assertTrue

class WorkWithBasicAuth2 {

    private val driver by lazy {
        FirefoxDriver()
    }

    @BeforeTest
    @Throws(Exception::class)
    fun setUp() {
        driver.get("http://admin:admin@the-internet.herokuapp.com/basic_auth")
    }

    @AfterTest
    @Throws(Exception::class)
    fun tearDown() {
        // Quit session (closes browser)
        driver.quit()
    }

    @Test
    fun workWithBasicAuthTest2() {
        driver.get("http://the-internet.herokuapp.com/basic_auth")
        val pageMessage = driver.findElement(By.cssSelector("p")).text
        assertTrue(pageMessage.contains("Congratulations!"))
    }
}